### ISSUE TITLE
OSError: .......cannot open shared object file: No such file or directory

### ISSUE THREAD

**Issue by Saurabh11811** [marella/ctransformers#209]
I am using the base code as given in HF 

from ctransformers import AutoModelForCausalLM
llm = AutoModelForCausalLM.from_pretrained('marella/gpt-2-ggml')
print(llm('AI is going to'))

While this works fine locally on Mac machine, it fails on any of the linux environment (I tried Debian, Ubuntu, Alpine etc. ) using docker.

Here are the errors for each of them

******************************************
For Apline (FROM python:3.12.1-alpine3.19)

Error is - OSError: Error relocating /usr/local/lib/python3.12/site-packages/ctransformers/lib/basic/libctransformers.so: unsupported relocation type 7

******************************************

For both Debian, Ubuntu 
Error is - OSError: /usr/local/lib/python3.11/site-packages/ctransformers/lib/basic/libctransformers.so: cannot open shared object file: No such file or directory

********************************************
Even for direct python3.11 images same error comes. 

I am trying the latest version

**Comment by jash-gmu** [comment]
i am facing the same issue, did you find any solution ?

**Comment by Saurabh11811** [comment]
Yes. The issue is because of CPU architecture type for the pulled image. I was using mac and docker by default pull linux image of arm type which is incompatible with ctransformers unless you compile from source (no clear steps). Workaround is you pull the x86/amd image in docker by defining platform parameter in docker file and it will work. BrSaurabhSent from my iPhoneOn 23 Apr 2024, at 8:12 PM, Jashwanth Raj Gowlikar ***@***.***> wrote:﻿
i am facing the same issue, did you find any solution ?

**Comment by Ajayvenki** [comment]
Try. https://github.com/marella/ctransformers/issues/9
